src/components/product-tiles/ProductTiles.test.tsx

```diff
@@ -1,18 +1,77 @@
+import { rest } from 'msw';
 import { render, screen } from '../../test-utils';
 import { registerIcons } from '../../registerIcons';
+import { server } from '../../mocks/server';
+import { API_GET_PRODUCTS } from '../../constants/api';
+import { mockProductsResponse } from '../../mocks/mockProductResponse';
 import { ProductTiles } from './ProductTiles';
 
-// TODO: Add tests
-// Does it render a list of tiles with the correct properties?
-// Does it render the an alert component with the correct text if there is an error?
-// Does it render a loading spinner while the list of products is being fetched?
-
-test('renders a list with the corrent number of product tiles', async () => {
+test('renders a list with the correct number of product tiles', async () => {
   registerIcons();
+  const numberOfMockTiles = mockProductsResponse.products.length;
   render(<ProductTiles />);
   const productTiles = await screen.findByTestId('product-tiles-container');
   expect(productTiles).toBeInTheDocument();
 
   const tiles = await screen.findAllByTestId('product-tile');
-  expect(tiles).toHaveLength(2);
+  expect(tiles).toHaveLength(numberOfMockTiles);
+});
+
+test('renders a list of tiles with correct properties', async () => {
+  registerIcons();
+  const numberOfMockTiles = mockProductsResponse.products.length;
+  render(<ProductTiles />);
+
+  const productTiles = await screen.findAllByTestId('product-tile');
+  const firstTileName = await screen.findByText(
+    `${mockProductsResponse.products[0]._source.name}`
+  );
+  const secondTileName = await screen.findByText(
+    `${mockProductsResponse.products[1]._source.name}`
+  );
+  const tilePrices = await screen.findAllByText(/£/);
+  const tileImages = await screen.findAllByRole('img');
+  const tileButtons = await screen.findAllByRole('button', {
+    name: /add to cart/i,
+  });
+
+  expect(productTiles).toHaveLength(numberOfMockTiles);
+  expect(firstTileName).toBeInTheDocument();
+  expect(secondTileName).toBeInTheDocument();
+  expect(tilePrices).toHaveLength(numberOfMockTiles);
+  expect(tileImages).toHaveLength(numberOfMockTiles);
+  expect(tileButtons).toHaveLength(numberOfMockTiles);
+});
+
+test("Renders alert component with correct text when there's an error", async () => {
+  server.resetHandlers(
+    rest.get(`${API_GET_PRODUCTS}`, (req, res, ctx) => res(ctx.status(500)))
```

Comment:
Nice! I was going to say not to do this here. But I forgot about the error test. Well done :)

Reply:
Yes it took me a bit to work out where the error needed to go! Thanks